activesupport/lib/active_support/broadcast_logger.rb

```diff
@@ -0,0 +1,207 @@
+  #
+  # Modifying the log level for all broadcasted loggers.
+  #
+  #   stdout_logger = Logger.new(STDOUT)
+  #   file_logger   = Logger.new("development.log")
+  #   broadcast = BroadcastLogger.new(stdout_logger, file_logger)
+  #
+  #   broadcast.level = Logger::FATAL # Modify the log level for the whole broadcast.
+  #
+  # Stop broadcasting log to a sink.
+  #
+  #   stdout_logger = Logger.new(STDOUT)
+  #   file_logger   = Logger.new("development.log")
+  #   broadcast = BroadcastLogger.new(stdout_logger, file_logger)
+  #   broadcast.info("Hello world!") # Writes the log to STDOUT and the development.log file.
+  #
+  #   broadcast.stop_broadcasting_to(file_logger)
+  #   broadcast.info("Hello world!") # Writes the log *only* to STDOUT.
+  #
+  # At least one sink has to be part of the broadcast. Otherwise, your logs will not
+  # be written anywhere. For instance:
+  #
+  #   broadcast = BroadcastLogger.new
+  #   broadcast.info("Hello world") # The log message will appear nowhere.
+  class BroadcastLogger
+    include ActiveSupport::LoggerSilence
+
+    # @return [Array<Logger>] All the logger that are part of this broadcast.
+    attr_reader :broadcasts
+    attr_reader :formatter
+
+    def initialize(*loggers)
+      @broadcasts = []
+
+      broadcast_to(*loggers)
+    end
+
+    # Add logger(s) to the broadcast.
+    #
+    # @param loggers [Logger] Loggers that will be part of this broadcast.
+    #
+    # @example Broadcast yours logs to STDOUT and STDERR
+    #   broadcast.broadcast_to(Logger.new(STDOUT), Logger.new(STDERR))
+    def broadcast_to(*loggers)
+      @broadcasts.concat(loggers)
+    end
+
+    # Remove a logger from the broadcast. When a logger is removed, messages sent to
+    # the broadcast will no longer be written to its sink.
+    #
+    # @param logger [Logger]
+    def stop_broadcasting_to(logger)
+      @broadcasts.delete(logger)
+    end
+
+    # Implemented for duck typing.
+    def progname; end
+
+    # Implemented for duck typing.
+    def level; end
+
+    def <<(message)
+      dispatch { |logger| logger.<<(message) }
+    end
+
+    def add(*args, &block)
+      dispatch { |logger| logger.add(*args, &block) }
+    end
+    alias_method :log, :add
+
+    def debug(*args, &block)
+      dispatch { |logger| logger.debug(*args, &block) }
+    end
+
+    def info(*args, &block)
+      dispatch { |logger| logger.info(*args, &block) }
+    end
+
+    def warn(*args, &block)
+      dispatch { |logger| logger.warn(*args, &block) }
+    end
+
+    def error(*args, &block)
+      dispatch { |logger| logger.error(*args, &block) }
+    end
+
+    def fatal(*args, &block)
+      dispatch { |logger| logger.fatal(*args, &block) }
+    end
+
+    def unknown(*args, &block)
+      dispatch { |logger| logger.unknown(*args, &block) }
+    end
+
+    def formatter=(formatter)
+      dispatch { |logger| logger.formatter = formatter }
+
+      @formatter = formatter
+    end
+
+    def progname=(progname)
```

Comment:
Do we want to allow assigning the progname? I think all the loggers inside the broadcast will have different prog names and we don't want to make all of them the same.

Reply:
Yeah I agree that assigning the progname for all loggers is weird. But I was thinking that the behaviour on this broadcast should make it clear that any methods called on it will modify all inner loggers. Could be confusing if `progname=` is the only exception to that rule?

No strong opinion though and I'm happy to change it.

Reply:
Good point. But I think progname is really uniq for each logger so I think it would be better to now allow a mass update. We can even raise an exception. But given `progname` will return `Boardcast` make this method change only the progname of the broadcast?